pkg/handlers/embedded_cluster_confirm_cluster_management.go

```diff
@@ -0,0 +1,95 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/kotsutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/preflight"
+	"github.com/replicatedhq/kots/pkg/store"
+	storetypes "github.com/replicatedhq/kots/pkg/store/types"
+)
+
+type ConfirmEmbeddedClusterManagementResponse struct {
+	VersionStatus string `json:"versionStatus"`
+}
+
+func (h *Handler) ConfirmEmbeddedClusterManagement(w http.ResponseWriter, r *http.Request) {
+	apps, err := store.GetStore().ListInstalledApps()
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to list installed apps: %w", err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if len(apps) == 0 {
+		logger.Error(fmt.Errorf("no installed apps found"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	app := apps[0]
+
+	downstreamVersions, err := store.GetStore().FindDownstreamVersions(app.ID, true)
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to find downstream versions: %w", err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if len(downstreamVersions.PendingVersions) == 0 {
+		logger.Error(fmt.Errorf("no pending versions found"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	pendingVersion := downstreamVersions.PendingVersions[0]
+
+	if pendingVersion.Status == storetypes.VersionPendingClusterManagement {
```

Comment:
what do you think if we return an error if the version status is something different?

Reply:
added 400 response as this endpoint should not be called if the version is not `pending_cluster_management`